src/utils/useWeb3Connect.js

```diff
@@ -443,10 +456,9 @@ function useWeb3Connect() {
           continue;
         }
         console.log({ hash });
-        let proposal = await getThreadFirstPost(hash);
+        let proposal = (await getThreadFirstPost(hash)).message;
         proposal.id = i;
-        console.log({ proposal });
-        tempProposals.push(proposal.message);
```

Comment:
This was a bug, it wasn't taking all the other values of the proposal such as the `id` which meant it was null when you tried to vote. See the fix on line 459 in the new version.